adapters/repos/db/index.go

```diff
@@ -566,6 +567,17 @@ func (i *Index) putObjectBatch(ctx context.Context, objects []*storobj.Object,
 		wg.Add(1)
 		go func(shardName string, group objsAndPos) {
 			defer wg.Done()
+
+			defer func() {
+				err := recover()
+				if err != nil {
+					for pos := range group.pos {
+						out[pos] = fmt.Errorf("an unexpected error occurred: %s", err)
+					}
+					fmt.Printf("panic: %s\n", err)
```

Comment:
Not sure if it matters, but I noticed that the go stdlib tends to write to stderr rather than stdout in this case:
```go
fmt.Fprintf(os.Stderr, "panic: %s\n", err)
```

Reply:
Great point. Adjusted both.